### ISSUE TITLE
java.lang.IncompatibleClassChangeError when unmarshaling Json

### ISSUE THREAD

**Issue by itaross** [RustedBones/akka-http-scalapb#9]
Hi @RustedBones, can you please provide support for this error?
Thanks

```
2020.05.09 15:28:05 ERROR Uncaught error from thread [ArgoApi-akka.actor.default-dispatcher-10]: Found class com.google.protobuf.descriptor.FieldDescriptorProto$Type, but interface was expected, shutting down JVM since 'akka.jvm-exit-on-fatal-error' is enabled for ActorSystem[ArgoApi]
2020.05.09 15:28:05 ERROR java.lang.IncompatibleClassChangeError: Found class com.google.protobuf.descriptor.FieldDescriptorProto$Type, but interface was expected
2020.05.09 15:28:05 ERROR 	at scalapb.json4s.Printer.$anonfun$toJson$1(JsonFormat.scala:243)
2020.05.09 15:28:05 ERROR 	at scala.collection.immutable.Vector.foreach(Vector.scala:1794)
2020.05.09 15:28:05 ERROR 	at scalapb.json4s.Printer.toJson(JsonFormat.scala:241)
2020.05.09 15:28:05 ERROR 	at fr.davit.akka.http.scaladsl.marshallers.scalapb.ScalaPBJsonSupport.$anonfun$scalaPBJsonMarshaller$1(ScalaPBJsonSupport.scala:48)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.marshalling.Marshaller.$anonfun$compose$2(Marshaller.scala:73)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.marshalling.Marshaller$$anon$1.apply(Marshaller.scala:92)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.marshalling.PredefinedToResponseMarshallers.$anonfun$fromStatusCodeAndHeadersAndValue$2(PredefinedToResponseMarshallers.scala:59)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.marshalling.Marshaller$$anon$1.apply(Marshaller.scala:92)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.marshalling.Marshaller.$anonfun$compose$2(Marshaller.scala:73)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.marshalling.Marshaller$$anon$1.apply(Marshaller.scala:92)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.marshalling.Marshal.toResponseFor(Marshal.scala:55)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.marshalling.ToResponseMarshallable.apply(ToResponseMarshallable.scala:17)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.marshalling.ToResponseMarshallable.apply$(ToResponseMarshallable.scala:16)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.marshalling.ToResponseMarshallable$$anon$1.apply(ToResponseMarshallable.scala:22)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.RequestContextImpl.complete(RequestContextImpl.scala:42)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.directives.RouteDirectives.$anonfun$complete$1(RouteDirectives.scala:47)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.StandardRoute$$anon$1.apply(StandardRoute.scala:19)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.StandardRoute$$anon$1.apply(StandardRoute.scala:19)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.directives.BasicDirectives.$anonfun$mapRouteResult$2(BasicDirectives.scala:68)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.directives.FutureDirectives.$anonfun$onComplete$3(FutureDirectives.scala:37)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.util.FastFuture$.$anonfun$transformWith$1(FastFuture.scala:37)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.util.FastFuture$.strictTransform$1(FastFuture.scala:41)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.util.FastFuture$.transformWith$extension(FastFuture.scala:45)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.util.FastFuture$.transformWith$extension(FastFuture.scala:37)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.directives.FutureDirectives.$anonfun$onComplete$2(FutureDirectives.scala:37)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.directives.BasicDirectives.$anonfun$textract$2(BasicDirectives.scala:161)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.directives.BasicDirectives.$anonfun$mapRouteResult$2(BasicDirectives.scala:68)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.directives.BasicDirectives.$anonfun$textract$2(BasicDirectives.scala:161)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.directives.BasicDirectives.$anonfun$mapRequestContext$2(BasicDirectives.scala:45)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.directives.BasicDirectives.$anonfun$textract$2(BasicDirectives.scala:161)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.RouteConcatenation$RouteWithConcatenation.$anonfun$$tilde$2(RouteConcatenation.scala:47)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.util.FastFuture$.strictTransform$1(FastFuture.scala:41)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.util.FastFuture$.transformWith$extension(FastFuture.scala:45)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.util.FastFuture$.flatMap$extension(FastFuture.scala:26)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.RouteConcatenation$RouteWithConcatenation.$anonfun$$tilde$1(RouteConcatenation.scala:44)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.directives.BasicDirectives.$anonfun$mapRequestContext$2(BasicDirectives.scala:45)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.directives.BasicDirectives.$anonfun$textract$2(BasicDirectives.scala:161)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.RouteConcatenation$RouteWithConcatenation.$anonfun$$tilde$2(RouteConcatenation.scala:47)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.util.FastFuture$.strictTransform$1(FastFuture.scala:41)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.util.FastFuture$.transformWith$extension(FastFuture.scala:45)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.util.FastFuture$.flatMap$extension(FastFuture.scala:26)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.RouteConcatenation$RouteWithConcatenation.$anonfun$$tilde$1(RouteConcatenation.scala:44)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.directives.ExecutionDirectives.$anonfun$handleExceptions$2(ExecutionDirectives.scala:32)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.directives.BasicDirectives.$anonfun$mapRequestContext$2(BasicDirectives.scala:45)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.directives.BasicDirectives.$anonfun$textract$2(BasicDirectives.scala:161)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.RouteConcatenation$RouteWithConcatenation.$anonfun$$tilde$2(RouteConcatenation.scala:47)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.util.FastFuture$.strictTransform$1(FastFuture.scala:41)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.util.FastFuture$.transformWith$extension(FastFuture.scala:45)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.util.FastFuture$.flatMap$extension(FastFuture.scala:26)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.RouteConcatenation$RouteWithConcatenation.$anonfun$$tilde$1(RouteConcatenation.scala:44)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.directives.BasicDirectives.$anonfun$textract$2(BasicDirectives.scala:161)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.directives.BasicDirectives.$anonfun$mapRouteResultWith$2(BasicDirectives.scala:74)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.directives.BasicDirectives.$anonfun$textract$2(BasicDirectives.scala:161)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.directives.ExecutionDirectives.$anonfun$handleExceptions$2(ExecutionDirectives.scala:32)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.directives.BasicDirectives.$anonfun$textract$2(BasicDirectives.scala:161)
2020.05.09 15:28:05 ERROR 	at akka.http.scaladsl.server.Route$.$anonfun$asyncHandler$1(Route.scala:86)
2020.05.09 15:28:05 ERROR 	at akka.stream.impl.fusing.MapAsyncUnordered$$anon$31.onPush(Ops.scala:1401)
2020.05.09 15:28:05 ERROR 	at akka.stream.impl.fusing.GraphInterpreter.processPush(GraphInterpreter.scala:541)
2020.05.09 15:28:05 ERROR 	at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:423)
2020.05.09 15:28:05 ERROR 	at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:625)
2020.05.09 15:28:05 ERROR 	at akka.stream.impl.fusing.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:502)
2020.05.09 15:28:05 ERROR 	at akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:600)
2020.05.09 15:28:05 ERROR 	at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:769)
2020.05.09 15:28:05 ERROR 	at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:784)
2020.05.09 15:28:05 ERROR 	at akka.actor.Actor.aroundReceive(Actor.scala:535)
2020.05.09 15:28:05 ERROR 	at akka.actor.Actor.aroundReceive$(Actor.scala:533)
2020.05.09 15:28:05 ERROR 	at akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:691)
2020.05.09 15:28:05 ERROR 	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:575)
2020.05.09 15:28:05 ERROR 	at akka.actor.ActorCell.invoke(ActorCell.scala:545)
2020.05.09 15:28:05 ERROR 	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:270)
2020.05.09 15:28:05 ERROR 	at akka.dispatch.Mailbox.run(Mailbox.scala:231)
2020.05.09 15:28:05 ERROR 	at akka.dispatch.Mailbox.exec(Mailbox.scala:243)
2020.05.09 15:28:05 ERROR 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
2020.05.09 15:28:05 ERROR 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
2020.05.09 15:28:05 ERROR 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
2020.05.09 15:28:05 ERROR 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
2020.05.09 15:28:05 ERROR 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)

```

**Comment by RustedBones** [comment]
Hi @itaross.
From what I read, this comes from a breaking change from `scalaPB`. You are probably using version `0.10.+`, See [here](https://github.com/scalapb/ScalaPB/blob/master/CHANGELOG.md#0100).

They change some definitions from `trait` to `abstract class`.

I'm going to prepare a new release to fix that. Thanks for reporting the issue!

**Comment by itaross** [comment]
Thanks for the reply, for now I have solved implementing my own unmarshaller.
Looking forward for your update.

**Comment by RustedBones** [comment]
Just published version `0.2.2` with support of scalaPB `0.10`.

I noticed that the `scalapb-json4s` dependency must be a least  `0.10.1` to avoid the above exception.  Release `0.10.0` is not compiled against the good scalapb runtime. If you don't override this dependency, everything should be fine.

Try it out and let me know if it is solved !
